### PR TITLE
Use "Inverted" terminology for cam options, add "Restore Original" option

### PR DESCRIPTION
**Pull request by dallmeyer**
This reverts back to using "Inverted" instead of "Flipped" terminology for camera controls (and adjusts the related code accordingly). In order to avoid confusing players who are used to most of these controls being inverted in the original Jak game, it defaults to the original controls and I also added an option to restore these controls:
- first-person horizontal: NOT inverted
- first-person vertical: inverted
- third person horizontal: inverted
- third-person vertical: inverted

@ManDude I think this is more in line with what you were thinking?

Below video shows the default Jak 1 values, then I toggle them all and use the "Restore Original Game Controls" option to reset

https://user-images.githubusercontent.com/2515356/169888688-0d8302ba-bdf6-4c8a-85b5-8b54d2bab6d6.mp4

### CODE REVIEW

File: decompiler/config/all-types.gc [hunk 1]
```diff
@@ -1307,11 +1307,12 @@
   ;; extra IDs for pc port
   (camera-options               #x1000)
   (normal                       #x1001)
-  (flipped                      #x1002)
+  (inverted                     #x1002)
   (camera-controls-first-horz   #x1003)
   (camera-controls-first-vert   #x1004)
   (camera-controls-third-horz   #x1005)
   (camera-controls-third-vert   #x1006)
+  (camera-controls-original     #x1007)
```
Comment (ManDude) [inline decompiler/config/all-types.gc:1315]: Good thing I future proofed some of these IDs!

**Review by ManDude — APPROVED** [review]
Looks fine to me.

**Comment by water111** [comment]
Everything looks good to me, but there's a merge conflict. If you can fix that, I'll merge this.

File: goal_src/pc/progress-pc.gc [hunk 2]
```diff
@@ -899,7 +900,11 @@
                   (load-level-text-files (-> *level-task-data* (-> obj display-level-index) text-group-index))
                   (set! (-> obj next-display-state) (progress-screen invalid))
                   )
+                 (((game-text-id camera-controls-original))
+                  ;; restore default camera controls
+                  (reset-original-camera *pc-settings*)
```
Comment (ManDude) [inline goal_src/pc/progress-pc.gc:905]: Add a `(sound-play-by-name (static-sound-name "cursor-options") (new-sound-id) 1024 0 0 1 #t)` so there's an audio confirmation when you select the option. This should (but I'm not sure) correspond to that "ding" sound.

**Review by ManDude — CHANGES_REQUESTED** [review]
Looks good but I just want some code to be cleaned up, and some sound effects for UX!

File: goal_src/engine/camera/cam-states.gc [hunk 3]
```diff
@@ -2329,10 +2329,10 @@
...
     )
     ;; changed for pc port, only used for Third-Person Vertical camera
-    (#if PC_PORT
-        (* (if (-> *pc-settings* third-camera-vflip?) -1.0 1.0) f0-0)
-        f0-0
-        )
+    (*
+      (if (and PC_PORT (-> *pc-settings* third-camera-v-inverted?)) 1.0 -1.0)
+      f0-0
+    )
```
[the remark this answers is not in the document]
Comment (dallmeyer) [reply goal_src/engine/camera/cam-states.gc:2335]: Ah okay I noticed the refactoring but wasn't sure the motivation behind it - makes sense, will update

**Pull request merged**